sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -1036,9 +1036,11 @@ namespace Azure { namespace Storage { namespace Blobs {
         {
           throw std::runtime_error("HTTP status code " + std::to_string(http_status_code));
         }
-        XmlReader reader(
-            reinterpret_cast<const char*>(http_response.GetBodyBuffer().data()),
-            http_response.GetBodyBuffer().size());
+
+        auto stream = http_response.GetBodyStream();
+        auto unique_buffer = Core::Http::Response::GetBodyBufferFromStream(stream);
+
+        XmlReader reader(reinterpret_cast<const char*>(unique_buffer.get()), stream->Length());
```

Comment:
@JinmingHu-MSFT @vinjiang @katmsft 
We though that removing the bodyBuffer from the HTTP Request and Response simplifies the transport a lot!.

So, I have removed bodyBuffer and let transport working and supporting only body stream in all cases.

Any service, like storage, can still use a body buffer on the client to build a body request, and when building the HTTP Response, it can be wrapped with a MemoryStream.
I have updated the MemoryStream to be able to create it taking an std::vector<uint8_t> as parameter.

See how this file got updated to move the body buffer you created to the memoryStream (instead of moving it to the HTTP Request).
Also, I added a convenient way to get an std::vector<uint8_t> created from an stream (it basically allocates the vector and read from the stream). 

Also, I removed the libcurl callbacks and let only the custom send/receive that allow us to read and write to a socket directly.

Let me know if you have any questions.

Reply:
> Also, I added a convenient way to get an std::vector<uint8_t> created from an stream (it basically allocates the vector and read from the stream).

I really don't feel the `MemoryBodyStream` should have it's own copy of data.

Reply:
> We though that removing the bodyBuffer from the HTTP Request and Response simplifies the transport a lot!.

This is great.

Reply:
> Also, I added a convenient way to get an std::vector<uint8_t> created from an stream (it basically allocates the vector and read from the stream).
As long as the memory buffer can be directly got from the stream without copying, I'm fine with the design.